Tests/DiligentToolsTest/src/RenderStateNotationParser/RenderStateNotationParserTest.cpp

```diff
@@ -597,6 +597,45 @@ TEST(Tools_RenderStateNotationParser, RenderStateNotationParserTest)
     });
 }
 
+TEST(Tools_RenderStateNotationParser, DuplicationResorcesTest)
+{
+    RefCntAutoPtr<IRenderStateNotationParser> pParser = LoadFromFile("DuplicationResources.json");
+    ASSERT_NE(pParser, nullptr);
+}
+
+TEST(Tools_RenderStateNotationParser, RedefinitionResourcesTest)
+{
+    {
+        TestingEnvironmentScope TestScope{
+            "Failed to parse file: 'RedefinitionResourceSignature.json'.",
+            "Redefinition of resource signature 'TestName'.",
+            "Redefinition of resource signature 'TestName'."};
+
+        RefCntAutoPtr<IRenderStateNotationParser> pParser = LoadFromFile("RedefinitionResourceSignature.json");
+        ASSERT_NE(pParser, nullptr);
```

Comment:
You can use `EXPECT_NE` here as the test can continue in case the parser is null